File: frontend/src/api/dish.ts
import axios from '@/utils/axios'

export interface Dish {
  id: number
  name: string
  imageUrl?: string
  description?: string
  status: number
  createTime: string
}

export interface DishAddRequest {
  name: string
  imageUrl?: string
  description?: string
}

export const dishApi = {
  list(): Promise<Dish[]> {
    return axios.get('/dish/list')
  },

  add(data: DishAddRequest): Promise<Dish> {
    return axios.post('/dish/add', data)
  },

  update(id: number, data: DishAddRequest): Promise<Dish> {
    return axios.put(`/dish/update/${id}`, data)
  },

  delete(id: number): Promise<void> {
    return axios.delete(`/dish/delete/${id}`)
  }
}

File: backend/src/main/java/com/example/mealplan/entity/Dish.java
package com.example.mealplan.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("dish")
public class Dish {
    @TableId(type = IdType.AUTO)
    private Long id;

    @TableField("name")
    private String name;

    @TableField("image_url")
    private String imageUrl;

    @TableField("description")
    private String description;

    @TableField("status")
    private Integer status;

    @TableField("create_time")
    private LocalDateTime createTime;
}